### PR TITLE
Add ability to disable E2EE

### PR DESCRIPTION
**Pull request by claucambra**
This PR adds the ability to disable E2EE through the account settings. It deletes encrypted files that are locally synced when E2EE is disabled.

https://user-images.githubusercontent.com/70155116/201427455-9f1a7ab4-dba7-461e-bedb-8f0d7be27c7d.mov

Part of https://github.com/nextcloud/desktop/issues/5126

### CODE REVIEW

**Review by mgallien — CHANGES_REQUESTED** [review]
`FolderManTest` test is crashing
can you have a look ?

**Comment by claucambra** [comment]
> `FolderManTest` test is crashing
> can you have a look ?

fixed

**Comment by mgallien** [comment]
> > `FolderManTest` test is crashing
> > can you have a look ?
> 
> fixed

I will try to have another look today

File: src/gui/folder.cpp [hunk 1]
```diff
@@ -1332,6 +1332,60 @@ void Folder::slotAboutToRemoveAllFiles(SyncFileItem::Direction dir, std::functio
     msgBox->open();
 }
 
+void Folder::removeLocalE2eFiles()
```
Comment (allexzander) [inline src/gui/folder.cpp:1335]: @claucambra We should use selective sync blacklist and rely on a sync engine to remove files that we can't use.

**Review by allexzander — CHANGES_REQUESTED** [review]

File: src/libsync/discovery.cpp [hunk 2]
```diff
@@ -217,7 +217,14 @@ void ProcessDirectoryJob::process()
         if (handleExcluded(path._target, e, isHidden))
             continue;
 
-        if (_queryServer == InBlackList || _discoveryData->isInSelectiveSyncBlackList(path._original)) {
+        const auto isEncryptedFolderButE2eIsNotSetup = e.serverEntry.isValid() && e.serverEntry.isE2eEncrypted &&
+            _discoveryData->_account->e2e() && !_discoveryData->_account->e2e()->_publicKey.isNull() && _discoveryData->_account->e2e()->_privateKey.isNull();
```
Comment (claucambra) [inline src/libsync/discovery.cpp:221]: I would put each of these on a separate line for readbility

File: src/gui/folder.cpp [hunk 3]
```diff
@@ -1378,47 +1379,28 @@ void Folder::removeLocalE2eFiles()
...
+    // same as in void FolderStatusModel::slotApplySelectiveSync()
+    // only start sync if blackList has changed
+    // database lists will get updated during discovery
+    const auto changes = (existingBlacklistSet - expandedBlacklistSet) + (expandedBlacklistSet - existingBlacklistSet);
+    if (!changes.isEmpty()) {
+        _journal.setSelectiveSyncList(SyncJournalDb::SelectiveSyncUndecidedList, QStringList());
+        if (isBusy()) {
+            slotTerminateSync();
+        }
+        foreach (const auto &it, changes) {
```
Comment (claucambra) [inline src/gui/folder.cpp:1398]: I would not use `foreach` in new code, `for(const auto &x : y)` is better

**Comment by mgallien** [comment]
@claucambra there are build issues in the drone check and sonarcloud analysis failed due to failing tests

**Comment by claucambra** [comment]
> @claucambra there are build issues in the drone check and sonarcloud analysis failed due to failing tests

Fixed

**Review by allexzander — APPROVED** [review]

**Pull request merged**